github/handlers/ExecuteViewSubmitHandler.ts

```diff
@@ -101,6 +102,51 @@ export class ExecuteViewSubmitHandler {
                         }
                     }
                     break;
+                case ModalsEnum.MERGE_PULL_REQUEST_VIEW:{
+                    if (user.id) {
+                        const { roomId } = await getInteractionRoomData(this.read.getPersistenceReader(), user.id);
+                        if (roomId) {
+                            let room = await this.read.getRoomReader().getById(roomId) as IRoom;
+                            const repository = view.state?.[ModalsEnum.REPO_NAME_INPUT]?.[ModalsEnum.REPO_NAME_INPUT_ACTION];
+                            const pullNumber = view.state?.[ModalsEnum.PULL_REQUEST_NUMBER_INPUT]?.[ModalsEnum.PULL_REQUEST_NUMBER_INPUT_ACTION];
+                            const commitTitle = view.state?.[ModalsEnum.PULL_REQUEST_COMMIT_TITLE_INPUT]?.[ModalsEnum.PULL_REQUEST_COMMIT_TITLE_ACTION];
+                            const commitMessage = view.state?.[ModalsEnum.PULL_REQUEST_COMMIT_MESSAGE_INPUT]?.[ModalsEnum.PULL_REQUEST_COMMIT_MESSAGE_ACTION];
+                            const mergeMethod = view.state?.[ModalsEnum.PULL_REQUEST_MERGE_METHOD_INPUT]?.[ModalsEnum.PULL_REQUEST_MERGE_METHOD_OPTION];
+                            let accessToken = await getAccessTokenForUser(this.read, user, this.app.oauth2Config);
+                            if(accessToken?.token){
+                                let mergeResponse = await mergePullRequest(this.http,repository,accessToken.token,pullNumber,commitTitle,commitMessage,mergeMethod);
+                                if(mergeResponse?.serverError){
```

Comment:
Can we be more helpful in the most common cases failed PR merge ?

Because of the simplicity of the wrapping  (our UI is less than github.com's webapp)  we may want to do just as much or more in this case.

Reply:
@Sing-Li  So the most common reason for not being able to merge would be 1) merge conflicts or 2) if the user is not authorized to push code in the repo. And in both the cases the message is usually displayed in modal. Incase there are merge conflicts or any other reason for not automatically merging, then it will mention on `pullDetailsModal` that its not mergeable. We can try to add a feature to resolve merge conflicts but it will be not be as ideal as the one github due to AppsEngine and UIKit Restrictions.